maltoolbox/model.py

```diff
@@ -864,4 +723,193 @@ def load_from_file(
             serialized_model = load_dict_from_json_file(filename)
         else:
             raise ValueError('Unknown file extension, expected json/yml/yaml')
-        return cls._from_dict(serialized_model, lang_classes_factory)
+        return cls._from_dict(serialized_model, lang_graph)
+
+
+class ModelAsset:
+    def __init__(self, name:str, lg_asset: LanguageGraphAsset):
+        self.name:str = name
+        self.id:Optional[int] = None
+        self.lg_asset:LanguageGraphAsset = lg_asset
+        self.defenses:dict = {}
+        self.extras:dict = {}
+        self.associations:list = []
+        self.attack_step_nodes:list = []
+        for step in self.lg_asset.attack_steps.values():
+            if step.type == 'defense':
+                self.defenses[step.name] = 1.0 if step.ttc and \
+                    step.ttc['name'] == 'Enabled' else 0.0
+
+
+    @cached_property
+    def type(self):
+        return self.lg_asset.name
+
+
+    def _to_dict(self):
+        """Get dictionary representation of the asset."""
+
+        logger.debug(
+            'Translating "%s"(%d) to dictionary.',
+            self.name,
+            self.id
+        )
+
+
+        asset_dict: dict[str, Any] = {
+            'name': self.name,
+            'type': self.type
+        }
+
+        # Only add non-default values for defenses to improve legibility of
+        # the model format
+        asset_dict['defenses'] = {}
+        for defense, defense_value in self.defenses.items():
+            lg_step = self.lg_asset.attack_steps[defense]
+            default_defval = 1.0 if lg_step.ttc and \
+                    lg_step.ttc['name'] == 'Enabled' else 0.0
+            if defense_value != default_defval:
+                asset_dict['defenses'][defense] = defense_value
+        if asset_dict['defenses'] == {}:
+            # Do not include an empty defenses dictionary
+            del asset_dict['defenses']
+
+        if self.extras != {}:
+            # Add optional metadata to dict
+            asset_dict['extras'] = self.extras
+
+        return {self.id: asset_dict}
+
+
+    @classmethod
+    def _from_dict(
+        cls,
+        asset_object: dict,
+        lang_graph: LanguageGraph,
+        ) -> ModelAsset:
+
+        lg_asset = lang_graph.assets[asset_object['type']]
+
+        asset = ModelAsset(
+            name = f"{asset_object['name']}",
+            lg_asset = lg_asset
+        )
+
+        if 'extras' in asset_object:
+            asset.extras = asset_object['extras']
+
+        for defense, defense_value in asset_object.get(
+                'defenses', {}).items():
+            asset.defenses[defense] = float(defense_value)
+
```

Comment:
```suggestion
        asset = ModelAsset(
            name = f"{asset_object['name']}",
            lg_asset = lg_asset,
            defenses = {defense: float(value) for defense, value in asset_object.get('defenses', {}).items()},
            extras = asset_object.get('extras'),
        )
```

Treating defenses and extras differently than properties that you can set via `__init__` args seems arbitrary.

Reply:
The reasoning is that they're not mandatory, but I do see your point. I also considered adding the defenses as a parameter for the construct because I do think there are many situations where it's most convenient to set the defenses then. So, I'll add both for completion, but the extras are highly optional.

Reply:
You can set the optional stuff in `__init__` as args with default values.